### PR TITLE
Cleanup of DNSProvider extension kind

### PR DESCRIPTION
**Pull request by MartinWeindel**
drop deployment of dns.grdener.cloud CRDs

**How to categorize this PR?**

/kind technical-debt

**What this PR does / why we need it**:
Cleanup of all remaining code blocks dealing with DNSProviders
- drop registration of DNSProvider resources
- drop deployment of dns.gardener.cloud CRDs
- remove dependency on github.com/gardener/external-dns-management

Note: `DNSProvider` must still be in the list of supported extension kinds to allow deletion of any controller registration specifying this kind.

**Special notes for your reviewer**:
Part of #5270

**Release note**:

```breaking operator
The `DNSProvider` extension kind was removed. Please make sure to remove any `ControllerRegistrations` that include the `DNSProvider` kind. If you are using the extension `shoot-dns-service`, make sure to deploy the dns-controller-manager by extending its `ControllerDeployment` (see [Deployment of DNS controller manager](https://github.com/gardener/gardener-extension-shoot-dns-service/blob/master/docs/installation/setup.md#deployment-of-dns-controller-manager)).
```

### CODE REVIEW

File: pkg/apis/core/validation/controllerregistration.go [hunk 1]
```diff
@@ -58,8 +57,8 @@ var SupportedExtensionKinds = sets.NewString(
 	extensionsv1alpha1.NetworkResource,
 	extensionsv1alpha1.OperatingSystemConfigResource,
 	extensionsv1alpha1.WorkerResource,
-	// TODO: drop this, once we have externalized DNSProvider functionality, see https://github.com/gardener/gardener/issues/5270
-	dnsv1alpha1.DNSProviderKind,
+	// TODO: drop this with v1.55 or later to still support removal of any controller registrations specifying resources of kind DNSProvider, see https://github.com/gardener/gardener/issues/5270
```
Comment (timebertt) [inline pkg/apis/core/validation/controllerregistration.go:60]: Can you make the release note breaking for operators and mention that they need to remove any `ControllerRegistrations` that include the `DNSProvider` kind?

File: charts/gardener/provider-local/extension/doc.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:generate ../../../../hack/generate-controller-registration.sh provider-local . v0.0.0 ../../../../example/provider-local/garden/base/controller-registration.yaml BackupBucket:local BackupEntry:local DNSProvider:local DNSRecord:local ControlPlane:local Infrastructure:local OperatingSystemConfig:local Worker:local
+//go:generate ../../../../hack/generate-controller-registration.sh provider-local . v0.0.0 ../../../../example/provider-local/garden/base/controller-registration.yaml BackupBucket:local BackupEntry:local DNSRecord:local ControlPlane:local Infrastructure:local OperatingSystemConfig:local Worker:local
```
Comment (timebertt) [inline charts/gardener/provider-local/extension/doc.go:15]: Can you remove the `DNSProvider` handling from the following places as well?

RBAC: https://github.com/gardener/gardener/blob/b182557e8acd4ffc4e59a4bf324e070c35c9907c/charts/gardener/provider-local/extension/templates/rbac.yaml#L43-L53

Docs: https://github.com/gardener/gardener/blob/master/docs/extensions/provider-local.md#dnsprovider

File: pkg/apis/core/validation/controllerregistration.go [hunk 3]
```diff
@@ -58,8 +57,8 @@ var SupportedExtensionKinds = sets.NewString(
...
+	"DNSProvider",
```
Comment (timebertt) [inline pkg/apis/core/validation/controllerregistration.go:61]: Can you drop `DNS{Provider,Entry,Onwer}` / `dns.gardener.cloud` references also from the following places?

https://github.com/gardener/gardener/blob/b182557e8acd4ffc4e59a4bf324e070c35c9907c/pkg/apis/core/types_controllerregistration.go#L48-L50
https://github.com/gardener/gardener/blob/b182557e8acd4ffc4e59a4bf324e070c35c9907c/pkg/apis/core/types_controllerregistration.go#L55-L57
https://github.com/gardener/gardener/blob/master/docs/extensions/dnsrecord.md
https://github.com/gardener/gardener/blob/master/docs/usage/trouble_shooting_guide.md
https://github.com/gardener/gardener/blob/b182557e8acd4ffc4e59a4bf324e070c35c9907c/test/framework/shootmigrationtest.go#L396

**Review by timebertt — COMMENTED** [review]
Thank you very much, well done!
Just a few cleanups remaining :)

File: pkg/apis/core/validation/controllerregistration.go (hunk 1 above)
Comment (MartinWeindel) [reply pkg/apis/core/validation/controllerregistration.go:60]: Done. Thanks for the hint.

File: charts/gardener/provider-local/extension/doc.go (hunk 2 above)
Comment (MartinWeindel) [reply charts/gardener/provider-local/extension/doc.go:15]: done

File: pkg/apis/core/validation/controllerregistration.go (hunk 3 above)
Comment (MartinWeindel) [reply pkg/apis/core/validation/controllerregistration.go:61]: done

**Review by timebertt — APPROVED** [review]
Awesome, thank you very much!
/lgtm

**Pull request merged**